test_workspace/Queue.bosatsu

```diff
@@ -85,14 +87,46 @@ q12 = empty.push(1).push(2)
 
 def samp(r)(fn): flat_map_Rand(r, fn)
 
+rand_int = int_range_Rand(128)
+
 rand_geo_List_Int: Rand[List[Int]] = (
-  rand_int = int_range_Rand(128)
   len <- geometric_Int.samp()
   lst = replicate_List(rand_int, len)
   sequence_Rand(lst)
 )
 
-rand_Queue_Int: Rand[Queue[Int]] = rand_geo_List_Int.map_Rand(from_List)
+queue_from_list = rand_geo_List_Int.map_Rand(from_List)
+
+def rand_Queue_depth(depth: Nat) -> Rand[Queue[Int]]:
+    recur depth:
+        case Zero: queue_from_list
+        case Succ(n):
+          smaller = rand_Queue_depth(n)
+          pop_rand = smaller.map_Rand(pop)
+          push_rand = prod_Rand(rand_int, smaller).map_Rand(((h, q)) -> push(q, h))
+          rev_rand = smaller.map_Rand(reverse_Queue)
+          one_of(pop_rand, [push_rand, rev_rand])
+
+rand_Queue_Int: Rand[Queue[Int]] = rand_Queue_depth(int_to_Nat(50))
```

Comment:
this is building a pretty giant tree perhaps... definitely the checks take a while to run.